### PR TITLE
Authentication branch

### PR DESCRIPTION
**Pull request by gorrelljd21**
For ticket [Jessie - Pass header to GraphQL resolvers through context#258](https://github.com/pgdevelopers/apprenticeship/issues/258)

Added in authentication code in server.go and schema.resolvers.go. Just need to run server.go to see it run. 

The code isn't very DRY, but I made a ticket to refactor

### CODE REVIEW

File: gqlgen/server.go [hunk 1]
```diff
@@ -19,11 +21,34 @@ func main() {
...
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			// put it in context
+			ctx := context.WithValue(r.Context(), "API-Key", r.Header.Get("X-Api-key"))
+
+			// and call the next with our new context
+			r = r.WithContext(ctx)
+
+			// fmt.Println("in middleware land")
+			next.ServeHTTP(w, r)
```
Comment (librairica) [inline gqlgen/server.go:51]:
```suggestion
		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			// put it in context
			ctx := context.WithValue(r.Context(), "API-Key", r.Header.Get("X-Api-key"))
			// and call the next with our new context
			r = r.WithContext(ctx)
			next.ServeHTTP(w, r)
```

File: gqlgen/graph/schema.resolvers.go [hunk 2]
```diff
@@ -30,14 +31,20 @@ func (r *mutationResolver) InsertQuote(ctx context.Context, input model.NewQuote
 
 	bufferResponse := bytes.NewBuffer(response)
 
+	stringKey := ctx.Value("API-Key").(string)
+
 	request, err := http.NewRequest("POST", "http://34.160.90.176:80/quote", bufferResponse)
-	request.Header.Set("X-Api-Key", "COCKTAILSAUCE")
+	request.Header.Set("X-Api-Key", stringKey)
 	// request.Header.Set("Content-Type", "application/json")
```
Comment (librairica) [inline gqlgen/graph/schema.resolvers.go]:
```suggestion

```

File: gqlgen/graph/schema.resolvers.go [hunk 3]
```diff
@@ -130,6 +157,7 @@ func (r *queryResolver) QuoteID(ctx context.Context, id string) (*model.Quote, e
 	}
 
 	return quoteById, nil
+
 }
```
Comment (librairica) [inline gqlgen/graph/schema.resolvers.go:186]:
```suggestion
	return quoteById, nil
}
```

File: main.go [hunk 4]
```diff
@@ -81,8 +81,10 @@ func manageHeader(c *gin.Context) bool {
 }
 
 func deleteQuote(c *gin.Context) {
+
 	id := c.Param("id")
```
Comment (librairica) [inline main.go]:
```suggestion
func deleteQuote(c *gin.Context) {
	id := c.Param("id")
```

**Review by librairica — APPROVED** [review]
This is good stuff. I really like how you are returning a status code and message for deleting quotes. One thing I noticed is that when I don't pass the header, I am not getting any helpful unauthorized messages. Probably the most confusing is when I try to create a quote while unauthorized because I get the quote and author back without an id. Unless you want to fix it on this branch, make separate tickets to return unauthorized messages.

**Comment by librairica** [comment]
> This is good stuff. I really like how you are returning a status code and message for deleting quotes. One thing I noticed is that when I don't pass the header, I am not getting any helpful unauthorized messages. Probably the most confusing is when I try to create a quote while unauthorized because I get the quote and author back without an id. Unless you want to fix it on this branch, make separate tickets to return unauthorized messages.

unauthorized delete:
Screen Shot 2022-10-19 at 1 23 28 PM

unauthorized create: 
Screen Shot 2022-10-19 at 1 23 58 PM

unauthorized random quote and quote by id:
Screen Shot 2022-10-19 at 1 24 41 PM

File: gqlgen/graph/schema.resolvers.go [hunk 5]
```diff
@@ -16,8 +16,28 @@ import (
...
+// 	}
+
+// 	if exists {
+// 		if  != stringKey {
+// 			return true
+// 		}
+// 	}
+// 	return false
+// }
+
```
Comment (ntschl) [inline gqlgen/graph/schema.resolvers.go]: You can probs remove this bruh

**Review by ntschl — APPROVED** [review]
Looks good to me! Can remove commented out code for cleanliness. 👌

**Pull request merged**